uefi/firmwarevolume.go

```diff
@@ -58,12 +59,13 @@ type FirmwareVolume struct {
 	Blocks []Block
 }
 
-// Summary prints a multi-line representation of a FirmwareVolume object
-func (fv FirmwareVolume) Summary() string {
-	var (
-		guidName, guidString string
-		ok                   bool
-	)
+// This is required to avoid recursion while marshalling json
+type jsonFirmwareVolume FirmwareVolume
+// MarshalJSON marshals the Firmware Volume into JSON
+func (fv FirmwareVolume) MarshalJSON() ([]byte, error) {
+	var guidName, guidString string
+	var ok bool
+
 	if guid, err := uuid.FromBytes(fv.FileSystemGUID[:]); err == nil {
```

Comment:
Why not just use the existing FileSystemGUID field? This seems needlessly complicated?

Reply:
I wanted to make this human readable.

Reply:
I should also keep the FileSystemGUID variable, instead of not marshalling it.